### PR TITLE
Implement timestamp rebase from Julian to Gregorian calendars

### PR DESCRIPTION
**Pull request by ttnghia**
Support rebasing day/microsecond timestamps from the legacy hybrid (Julian + Gregorian) calendar to Proleptic Gregorian calendar.

Merge checklist:
 * [X] Implementation
 * [X] Unit tests
 * [x] Test with spark-rapids (https://github.com/NVIDIA/spark-rapids/pull/9649, all integration tests pass)

### CODE REVIEW

**Comment by ttnghia** [comment]
build

File: src/main/cpp/src/datetime_rebase.cu [hunk 1]
```diff
@@ -93,6 +94,59 @@ std::unique_ptr<cudf::column> gregorian_to_julian_days(cudf::column_view const &
   return output;
 }
 
+// Convert a number of Julian days since epoch to local date in Julian calendar.
+// Follow the implementation of `julian_from_days` from
+// https://howardhinnant.github.io/date_algorithms.html
+__device__ __inline__ cuda::std::chrono::year_month_day julian_from_days(int32_t days) {
+  auto const z = days + 719470;
```
Comment (gerashegalov) [inline src/main/cpp/src/datetime_rebase.cu:101]: can you add comments about the hardcoded constants used in the conversion

Comment (ttnghia) [reply src/main/cpp/src/datetime_rebase.cu:101]: Sorry there are many constants in the formula (1153`, `1460`, `719470` etc), which can't be explained in some sentences. Even the original implementation (https://howardhinnant.github.io/date_algorithms.html) doesn't have any effective way to explain them. One needs to read through the entire documentation to understand what they are. That's why the `cuda/std` adoption of such formulae (in `detail/libcxx/include/chrono` header) also can't comment anything but has to refer to the original implementation website above.

**Review by gerashegalov — APPROVED** [review]
LGTM

**Pull request merged**